### PR TITLE
add llvm-bitcode-linker and llvm-tools bins to ci-rustc's sysroot

### PR DESCRIPTION
**Pull request by onur-ozkan**
https://github.com/rust-lang/rust/pull/130040 is [reverted](https://github.com/rust-lang/rust/pull/130292) because adding component binaries directly to the dist tarball of the compiler caused conflicts (see https://github.com/rust-lang/rust/issues/130291 and https://github.com/rust-lang/rustup/issues/4019). This PR solves the original problem without touching the dist tarball.

r? Kobzol

### CODE REVIEW

**Comment by onur-ozkan** [comment]
We should probably wait for the [try build](https://github.com/rust-lang/rust/pull/122709#issuecomment-2348520035) result before landing this.

**Comment by Kobzol** [comment]
This is cursed.. :)

File: src/bootstrap/src/core/build_steps/compile.rs [hunk 1]
```diff
@@ -1772,6 +1815,9 @@ impl Step for Assemble {
             if target_compiler.stage == builder.top_stage {
                 builder.info(&format!("Creating a sysroot for stage{stage} compiler (use `rustup toolchain link 'name' build/host/stage{stage}`)", stage=target_compiler.stage));
             }
+
+            maybe_install_llvm_bitcode_linker(target_compiler);
```
Comment (Kobzol) [inline src/bootstrap/src/core/build_steps/compile.rs:1819]: This means that the bitcode linker will be compiled using the downloaded CI rustc compiler, right?

Comment (onur-ozkan) [reply src/bootstrap/src/core/build_steps/compile.rs:1819]: Yes.

**Pull request merged**